### ISSUE TITLE
Files >2GB don't work due to 32bit truncation

### ISSUE THREAD

**Issue by GeorgA** [buzz/mediainfo.js#11]
Open_Buffer_Continue_GoTo_Get() returns a 64bit seek position, but the binding to JS makes a long (32bit) out of it. Thus seeking in large files doesn't work.  It gets obviuos when comparing the offsets from JS and native code:

native:
lseek(4, 2131533686, SEEK_SET)
lseek(4, 4261375884, SEEK_SET)

JS:
offset 2130943862 state 5 seekto -33591412

4261375884 is -33591412 as int32_t ;)

BTW: I tried to fix that on my own in the binding, but emscripten seems to have long term troubles under Ubuntu. Can you give an example of a "proven" build system? Thanks...

**Comment by buzz** [comment]
> Can you give an example of a "proven" build system?

I'm on Arch Linux, using the official emscripten SDK build. Don't know about Ubuntu. But should work as well.

What error messages do you receive?

**Comment by GeorgA** [comment]
It's a python error with a long stacktrace at compiling/linking mediainfojs.cpp. Seems like a python-lib compatibility issue or so.

In the meantime I've built the full zoo (llvm, clang, emscripten) from scratch. Building felt like days, but with these binaries, compiling mediainfo.js works under Ubuntu. 

After that I was able to dig around the problem. Not that I know much of emscripten and JS, it seems that there is no possibility to return an 64bit int. Possibly a double would work (as filesizes are less than 52bits for a while). After trying int64_t, I skipped the double-idea and used a proven approach: A private 64bit variable in class MediaInfoJs stores the full value of open_buffer_continue_goto_get. That returns the lower 32bit and a new call the upper. Then JS assembles the full value again:
`if  (seekTol==-1 && seekToh==-1)
                seekTo=-1;
            else if (seekTol<0)
                seekTo=seekTol+4294967296 + (seekToh*4294967296);
            else
                seekTo=seekTol + (seekToh*4294967296);`

Looks ugly but works very well :)

BTW: There seems another performance issue at least for some MXF-files in the example JS. A seek at the near file end can return less data that CHUNK_SIZE, in my case 560 bytes. So this chunk.length is assigned to l which is used as the length for the next seek/read block. For my MXF, unfortunately a few 100MB are then read in this tiny chunks, taking veeery long... Now I use always CHUNK_SIZE for file.slice, works (fast...) and shows no collateral problems under IE11, Chrome, Chromium, FF and Safari. I also reduced TOTAL_MEMORY to 512MB (enabling IE11 again). I've tried a lot of files but none triggered an out of memory exception, so I stayed at that setup.
